### PR TITLE
Replace fwlinks in csharp/programming-guide/classes-and-structs

### PR DESCRIPTION
**Pull request by dalesmithii**
Replaces all fwlinks with their appropriate url in the csharp/programming-guide/classes-and-structs folder and subfolders.

Relates to #3424

### CODE REVIEW

File: docs/csharp/programming-guide/classes-and-structs/how-to-implement-and-call-a-custom-extension-method.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ author: "BillWagner"
 ms.author: "wiwagn"
 ---
 # How to: Implement and Call a Custom Extension Method (C# Programming Guide)
-This topic shows how to implement your own extension methods for any type in the [.NET Framework Class Library](http://go.microsoft.com/fwlink/?LinkID=217856), or any other .NET type that you want to extend. Client code can use your extension methods by adding a reference to the DLL that contains them, and adding a [using](../../../csharp/language-reference/keywords/using-directive.md) directive that specifies the namespace in which the extension methods are defined.  
+This topic shows how to implement your own extension methods for any type in the [.NET Framework Class Library](https://msdn.microsoft.com/library/gg145045(v=VS.100).aspx), or any other .NET type that you want to extend. Client code can use your extension methods by adding a reference to the DLL that contains them, and adding a [using](../../../csharp/language-reference/keywords/using-directive.md) directive that specifies the namespace in which the extension methods are defined.
```
Comment (mairaw) [inline docs/csharp/programming-guide/classes-and-structs/how-to-implement-and-call-a-custom-extension-method.md]: Perhaps we could change this sentence to get rid of that msdn link. My suggestion would be to change it to this:
This topic shows how to implement your own extension methods for any .NET type.

File: docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ Optional parameters in ExampleMethod
 ## COM Interfaces  
  Named and optional arguments, along with support for dynamic objects and other enhancements, greatly improve interoperability with COM APIs, such as Office Automation APIs.  
   
- For example, the [AutoFormat](http://go.microsoft.com/fwlink/?LinkId=148201) method in the Microsoft Office Excel [Range](http://go.microsoft.com/fwlink/?LinkId=148196) interface has seven parameters, all of which are optional. These parameters are shown in the following illustration.  
+ For example, the [AutoFormat](https://msdn.microsoft.com/library/microsoft.office.interop.excel.range.autoformat(office.11).aspx) method in the Microsoft Office Excel [Range](https://msdn.microsoft.com/library/microsoft.office.interop.excel.range(VS.80).aspx) interface has seven parameters, all of which are optional. These parameters are shown in the following illustration.
```
Comment (mairaw) [inline docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md]: my recommendation would be to point to the latest version of the AutoFormat API:
https://msdn.microsoft.com/library/microsoft.office.interop.excel.range.autoformat(v=office.15).aspx

same for range:
https://msdn.microsoft.com/library/microsoft.office.interop.excel.range(v=vs.15).aspx

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks for jumping on that up-for-grab issue and helping out @dalesmithii! I have some suggestions for a few improvements on a couple of spots. We can help out making those fixes too, just let us know.

File: docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md (hunk 2 above)
Comment (dalesmithii) [reply docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md]: Sure thing, quick question though. What is the difference between v=office.15 and v=vs.15. For range, it looks like they go to the same page. If there is no significant difference, does it make sense to use v=office.15 for both links for consistency?

**Review by mairaw — APPROVED** [review]
Thank you so much for making the changes @dalesmithii and contributing to docs. I'll merge this as soon as builds are green.

**Pull request merged**